examples/preprocessing/plot_scaling_importance.py

```diff
@@ -1,106 +1,149 @@
 
-# Make a train/test split using 30% test size
 X_train, X_test, y_train, y_test = train_test_split(
-    features, target, test_size=0.30, random_state=RANDOM_STATE
+    X, y, test_size=0.30, random_state=42
 )
+scaled_X_train = scaler.fit_transform(X_train)
+
+# %%
+# Effect of rescaling on a KNeighbors model
+# =========================================
+#
+# For the sake of visualizing the decision boundary of a KNeighbors models, in
+# this section we select a subset of 2 features that have values with different
+# orders of magnitude.
+#
+# Keep in mind that using a subset of the features to train the model may likely
+# leave out predictive variables, resulting in a boundary decision that does not
+# represent the decisions nor the statistical performance of a model trained on
+# the full set of features.
 
-# Fit to data and predict using pipelined GNB and PCA
-unscaled_clf = make_pipeline(PCA(n_components=2), GaussianNB())
-unscaled_clf.fit(X_train, y_train)
-pred_test = unscaled_clf.predict(X_test)
+import matplotlib.pyplot as plt
+from sklearn.inspection import DecisionBoundaryDisplay
+from sklearn.neighbors import KNeighborsClassifier
 
-# Fit to data and predict using pipelined scaling, GNB and PCA
-std_clf = make_pipeline(StandardScaler(), PCA(n_components=2), GaussianNB())
-std_clf.fit(X_train, y_train)
-pred_test_std = std_clf.predict(X_test)
 
-# Show prediction accuracies in scaled and unscaled data.
-print("\nPrediction accuracy for the normal test dataset with PCA")
-print(f"{accuracy_score(y_test, pred_test):.2%}\n")
+X_plot = X[["proline", "hue"]].to_numpy()
+X_plot_scaled = StandardScaler().fit_transform(X_plot)
+clf = KNeighborsClassifier(n_neighbors=20)
 
-print("\nPrediction accuracy for the standardized test dataset with PCA")
-print(f"{accuracy_score(y_test, pred_test_std):.2%}\n")
 
-# Extract PCA from pipeline
-pca = unscaled_clf.named_steps["pca"]
-pca_std = std_clf.named_steps["pca"]
+def fit_and_plot_model(X_plot, y, clf, ax):
+    clf.fit(X_plot, y)
+    disp = DecisionBoundaryDisplay.from_estimator(
+        clf,
+        X_plot,
+        response_method="predict",
+        alpha=0.5,
+        ax=ax,
+    )
+    disp.ax_.scatter(X_plot[:, 0], X_plot[:, 1], c=y, s=20, edgecolor="k")
+    disp.ax_.set_xlim((X_plot[:, 0].min(), X_plot[:, 0].max()))
+    disp.ax_.set_ylim((X_plot[:, 1].min(), X_plot[:, 1].max()))
+    return disp.ax_
 
-# Show first principal components
-print(f"\nPC 1 without scaling:\n{pca.components_[0]}")
-print(f"\nPC 1 with scaling:\n{pca_std.components_[0]}")
 
-# Use PCA without and with scale on X_train data for visualization.
+fig, (ax1, ax2) = plt.subplots(ncols=2, figsize=(12, 6))
+
+fit_and_plot_model(X_plot, y, clf, ax1)
+ax1.set_xlabel("proline")
+ax1.set_ylabel("hue")
+ax1.set_title("KNN w/o scaling")
+
+fit_and_plot_model(X_plot_scaled, y, clf, ax2)
+ax2.set_xlabel("scaled proline")
+ax2.set_ylabel("scaled hue")
+ax2.set_title("KNN with scaling")
+
+plt.show()
```

Comment:
I think that we should only call `plt.show` after the last plot. Otherwise, it will block the execution when executing in a command line.